src/index.js

```diff
@@ -34,25 +35,40 @@ export default function immutableStateInvariantMiddleware(options = {}) {
       // Track before potentially not meeting the invariant
       tracker = track(state);
 
-      invariant(
-        !result.wasMutated,
-        BETWEEN_DISPATCHES_MESSAGE,
-        (result.path || []).join('.')
-      );
+      if (logToConsoleOnly) {
+        result.wasMutated && console.error(
+          BETWEEN_DISPATCHES_MESSAGE,
+          (result.path || []).join('.')
+        );
+      } else {
+        invariant(
+          !result.wasMutated,
+          BETWEEN_DISPATCHES_MESSAGE,
+          (result.path || []).join('.')
+        );
+      }
 
       const dispatchedAction = next(action);
       state = getState();
 
       result = tracker.detectMutations();
       // Track before potentially not meeting the invariant
       tracker = track(state);
-
-      result.wasMutated && invariant(
-        !result.wasMutated,
-        INSIDE_DISPATCH_MESSAGE,
-        (result.path || []).join('.'),
-        stringify(action)
-      );
+      
+      if (logToConsoleOnly) {
+        result.wasMutated && console.error(
+          INSIDE_DISPATCH_MESSAGE,
+          (result.path || []).join('.'),
+          stringify(action)
+        );
+      } else {
+        invariant(
```

Comment:
If you see the diff with the previous version, you'll see that before we were doing a check before calling invariant. This fixed a problem where we were calling `stringify(action)` unnecessarily even when the result was not mutated.